index.src.html

```diff
@@ -4133,6 +4136,40 @@
 : <dfn event for=ImageTrack>change</dfn>
 :: Fired at the {{ImageTrack}} when the {{ImageTrack/frameCount}} is altered.
 
+Resource Reclamation{#resource-reclamation}
+==============================================
+
+When resources are constrained, a User Agent might want to pro-actively reclaim
+codecs. This is particularly true in the case where hardware codecs are limited,
+and shared accross all web pages.
+
+A User Agent is allowed to [=reclaim a codec=], freeing the resources of a
+<dfn>reclaimable codec</dfn>, which is a defined as a codec that is either an
+[=inactive codec=] or a [=background codec=].
+
+An <dfn>inactive codec</dfn> is a codec that has not received a call to
+`encode()` or `decode()` in the past 1 second, and has no pending outputs.
```

Comment:
What's a pending output exactly? I've used that term before to mean that there is no input queued or sent to the codec for which we expect a future output, but that's a per-codec thing that's not easy to state both simply and precisely.

The main difficulty in definition is frame reordering; a codec might be waiting for a future `encode()` or `decode()` before it can output a frame it has already processed, yet `flush()` might not be sufficient to force the output. If you count that as 'inactive' then it's not so hard to define.

Reply:
You're right, this is tricky, and I hadn't thought of reordered frames.

I've tentatively updated the wording to "and has not called its `output()` callback in the past 1 second"... I'm not sure if this is what we want, but the definition as a whole makes some sense to me at least now. "No inputs or outputs for 1 second".